src/bin/commands/containers.rs

```diff
@@ -0,0 +1,502 @@
+            Volumes::Prune(args) => args.run(engine),
+            Volumes::Create(args) => args.run(engine, toolchain),
+            Volumes::Remove(args) => args.run(engine, toolchain),
+        }
+    }
+
+    pub fn engine(&self) -> Option<&str> {
+        match self {
+            Volumes::List(l) => l.engine.as_deref(),
+            Volumes::RemoveAll(l) => l.engine.as_deref(),
+            Volumes::Prune(l) => l.engine.as_deref(),
+            Volumes::Create(l) => l.engine.as_deref(),
+            Volumes::Remove(l) => l.engine.as_deref(),
+        }
+    }
+
+    pub fn verbose(&self) -> bool {
+        match self {
+            Volumes::List(l) => l.verbose,
+            Volumes::RemoveAll(l) => l.verbose,
+            Volumes::Prune(l) => l.verbose,
+            Volumes::Create(l) => l.verbose,
+            Volumes::Remove(l) => l.verbose,
+        }
+    }
+}
+
+#[derive(Args, Debug)]
+pub struct ListContainers {
+    /// Provide verbose diagnostic output.
+    #[clap(short, long)]
+    pub verbose: bool,
+    /// Container engine (such as docker or podman).
+    #[clap(long)]
+    pub engine: Option<String>,
+}
+
+impl ListContainers {
+    pub fn run(self, engine: docker::Engine) -> cross::Result<()> {
+        list_containers(self, &engine)
+    }
+}
+
+#[derive(Args, Debug)]
+pub struct RemoveAllContainers {
+    /// Provide verbose diagnostic output.
+    #[clap(short, long)]
+    pub verbose: bool,
+    /// Force removal of containers.
+    #[clap(short, long)]
+    pub force: bool,
+    /// Remove containers. Default is a dry run.
+    #[clap(short, long)]
+    pub execute: bool,
+    /// Container engine (such as docker or podman).
+    #[clap(long)]
+    pub engine: Option<String>,
+}
+
+impl RemoveAllContainers {
+    pub fn run(self, engine: docker::Engine) -> cross::Result<()> {
+        remove_all_containers(self, &engine)
+    }
+}
+
+#[derive(Subcommand, Debug)]
+pub enum Containers {
+    /// List cross containers in local storage.
+    List(ListContainers),
+    /// Stop and remove cross containers in local storage.
+    RemoveAll(RemoveAllContainers),
+}
+
+impl Containers {
+    pub fn run(self, engine: docker::Engine) -> cross::Result<()> {
+        match self {
+            Containers::List(args) => args.run(engine),
+            Containers::RemoveAll(args) => args.run(engine),
+        }
+    }
+
+    pub fn engine(&self) -> Option<&str> {
+        match self {
+            Containers::List(l) => l.engine.as_deref(),
+            Containers::RemoveAll(l) => l.engine.as_deref(),
+        }
+    }
+
+    pub fn verbose(&self) -> bool {
+        match self {
+            Containers::List(l) => l.verbose,
+            Containers::RemoveAll(l) => l.verbose,
+        }
+    }
+}
+
+fn get_cross_volumes(engine: &docker::Engine, verbose: bool) -> cross::Result<Vec<String>> {
+    let stdout = docker::subcommand(engine, "volume")
+        .arg("list")
+        .arg("--format")
+        .arg("{{.Name}}")
```

Comment:
```suggestion
        .args(&["--format", "{{.Name}}"])
```

this could also be `.args(["foo", "bar"])` but we're using slices everywhere else so let's not change it. This is since 1.53.0 where `[T]: IntoIterator<Item=T>` now